### DISCUSSION TITLE
Remove spaces from the filenames of images saved using savemedia addon

### DISCUSSION THREAD

**Discussion by arunnbabu81** [buggyj/tiddlyclip-plugin#126] [Ideas]
https://talk.tiddlywiki.org/t/is-it-a-feasible-to-create-a-markdown-convertor-viewtoolbar-button-based-on-the-markdown-export-plug-in/7851/15

In this discussion, Mark was telling about the need for avoiding spaces in filenames. 
Images snaped using tiddlyclip and saved using savemedia add-on has spaces in the title (date part of the title) . See this example.

`Tweet-ASK_MSK-septo-optic dysplasia2023_2_24 18_09_51.png`

Here is a sample tiddlyclip rule used for snaps.

![image](https://github.com/buggyj/tiddlyclip-plugin/assets/67494083/f028ae72-abd2-4082-8cc1-15d794912ab0)

Is it possible to avoid spaces form filenames when using savemedia addon

**Comment by buggyj** [comment]
I need to see an example wiki

**Comment by buggyj** [comment]
if you want to replace the whitespace in a string,

```
#astr=this is some text
#match=/(\s)/g
#subst=-
#astr=((*#astr/#match/#subst*))
//expect #astr to be this-is-some-text
```

**Reply by arunnbabu81** [reply]
@buggyj Sorry for the delay. I busy was making another template wiki for my use.

https://sidetabs-wiki.tiddlyhost.com/#Tweet-danatsouza-splenic%20hemangiomas
This is an example tiddler.
![image](https://github.com/buggyj/tiddlyclip-plugin/assets/67494083/e9d7b38c-8fb0-40c7-b3c4-f80c9355d24c)
I want to see whether space can be removed from the filename of the image getting saved when  snap is taken using tiddlyclip and saved using savemedia addon

These are the snap rules I use - https://sidetabs-wiki.tiddlyhost.com/#newlocaluserSnaprule  and https://sidetabs-wiki.tiddlyhost.com/#newLocalDefaultclipSnaptitlerule

**Reply by buggyj** [reply]
your newlocaluserSnaprule would be

```
@text=((*@removelf(@text)*))
#astr=((*@userstring*))
#match=/(\s)/g
#subst=-
#astr=((*#astr/#match/#subst*))
$text=((*$text*)) \n\n((*@pageRef*))\n\n[img[((*@csaver(#astr,@snap)*))]]\n\n((*@text*))\n\n
```

**Reply by arunnbabu81** [reply]
@buggyj 

I use `title `(tiddler title) instead of the `userstring` in the snaprule. 
`#title=((*@pagetitle*))((*@dateTimeShort*)).((*#ext*))`

Can the code you provided be modified for using `title` instead of  `userstring`
Also how to remove spaces from `@dateTimeShort`

**Reply by buggyj** [reply]
you can replace @userstring with anything you like.

**Comment by arunnbabu81** [comment]
@buggyj What do you think about `spaces `in tiddler title ? Should spaces be avoided while creating tiddler title ? Instead should I use symbols like `_` in place of `space`

**Reply by buggyj** [reply]
Having spaces in file names can be problematic, and should be avoided. As Tiddler titles can be a 'permalink' they can form part of the url, and in this case the spaces will be escaped to '%20' and this, at least, is confusing.

**Reply by arunnbabu81** [reply]
Can the date `@dateTimeShort` added to the filename of the images saved by savemedia and tiddlyclip avoid `spaces `and `underscores` and instead just use continuous numbers like in UTC format ?

**Reply by buggyj** [reply]
you can use the now macro instead:


`#datetime=((*@now([UTC]YYYY0MM0DD0hh0mm0ssXXX)*))`

**Comment by arunnbabu81** [comment]
@buggyj 
I have a tiddler titled `tweet_T2/FLAIR-mismatch_ecalabr`

Snap saved using this title got a filename `[img[media/tweet_T2_FLAIR-mismatch_ecalabr20230914171243082.png]]`

How did `T2/FLAIR` became `T2_FLAIR` ?

**Reply by buggyj** [reply]
the character "/" is illegal in a file name.
